File: 1913b_sad.cpp
/**
 * @file 1913b_sad.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Swap and Delete(https://codeforces.com/problemset/problem/1913/B)
 * @version 0.1
 * @date 2024-03-15
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	short t;
	int oc, zc, i;
	string s;

	cin >> t;
	while (t--)
	{
		cin >> s;
		oc = count(s.begin(), s.end(), '1');
		zc = s.length() - oc;
		i = 0;
		while (i < s.length())
		{
			if (s[i] == '1' && zc > 0)
				zc--;
			else if (s[i] == '0' && oc > 0)
				oc--;
			else
				break;
			i++;
		}
		cout << s.length() - i << '\n';
	}

	return (EXIT_SUCCESS);
}

File: 1913b_sad.py
"""
@file: 1913b_sad.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Swap and Delete(https://codeforces.com/problemset/problem/1913/B)
@version: 1.0
@date: 2024-10-06

@copyright: Copyright (c) 2024
"""

from collections import Counter, defaultdict
from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    t = int(input())
    for _ in range(t):
        s = input()
        sm = defaultdict(int, Counter(s))
        im = {"1": "0", "0": "1"}
        for i in range(len(s)):
            if not sm[im[s[i]]]:
                print(len(s) - i)
                break
            sm[im[s[i]]] -= 1
        else:
            print(0)


if __name__ == "__main__":
    main()
